File: src/main/java/brs/grpc/handlers/BroadcastTransactionBytesHandler.java
package brs.grpc.handlers;

import brs.DependencyProvider;
import brs.Transaction;
import brs.TransactionProcessor;
import brs.grpc.GrpcApiHandler;
import brs.grpc.proto.BrsApi;

public class BroadcastTransactionBytesHandler implements GrpcApiHandler<BrsApi.TransactionBytes, BrsApi.TransactionBroadcastResult> {

    private final DependencyProvider dp;

    public BroadcastTransactionBytesHandler(DependencyProvider dp) {
        this.dp = dp;
    }

    @Override
    public BrsApi.TransactionBroadcastResult handleRequest(BrsApi.TransactionBytes transactionBytes) throws Exception {
        return BrsApi.TransactionBroadcastResult.newBuilder()
                .setNumberOfPeersSentTo(dp.transactionProcessor.broadcast(Transaction.parseTransaction(dp, transactionBytes.getTransactionBytes().toByteArray())))
                .build();
    }
}

File: src/main/java/brs/grpc/handlers/BroadcastTransactionBytesHandler.kt
package brs.grpc.handlers

import brs.DependencyProvider
import brs.Transaction
import brs.grpc.GrpcApiHandler
import brs.grpc.proto.BrsApi

class BroadcastTransactionBytesHandler(private val dp: DependencyProvider) : GrpcApiHandler<BrsApi.TransactionBytes, BrsApi.TransactionBroadcastResult> {
    override suspend fun handleRequest(transactionBytes: BrsApi.TransactionBytes): BrsApi.TransactionBroadcastResult {
        return BrsApi.TransactionBroadcastResult.newBuilder()
                .setNumberOfPeersSentTo(dp.transactionProcessor.broadcast(Transaction.parseTransaction(dp, transactionBytes.transactionBytes.toByteArray()))!!)
                .build()
    }
}
